api/v1alpha1/clusterextension_types.go

```diff
@@ -382,9 +382,11 @@ type ServiceAccountReference struct {
 	Name string `json:"name"`
 }
 
-// ConfigMapReference references a configMap
-type ConfigMapReference struct {
-	Name string `json:"name"`
+// ConfigSourceReference can references a configMap, a secret, plain text config
+type ConfigSourceReferences struct {
+	ConfigMapNames []string `json:"configMapNames,omitempty"`
+	SecretNames    []string `json:"secretNames,omitempty"`
+	TextConfigs    []string `json:"textConfigs,omitempty"`
 }
```

Comment:
What if you did something like a discriminated union here?

I'm thinking something like:
```go
type ConfigSourceType string

const (
  ConfigSourceTypeConfigMap = "ConfigMap"
  ConfigSourceTypeSecret = "Secret"
  ConfigSourceTypeInline = "Inline" 
)

// CEL validation here to ensure type + property correctness
type ConfigSource struct {
  // +required
  Type ConfigSourceType `json:"type"`
  // +optional
  ConfigMap *ConfigMapReference `json:"configMap,omitempty"`
  ...
}
```

Then line 149 could become:
```go
ConfigSources []ConfigSource
```

I think something like this might fit better with the existing conventions of the ClusterExtension API